### PR TITLE
Add warning if WF name will be too long for submission

### PR DESCRIPTION
**Pull request by srimanob**
#### PR description:
The character length of computing workflow is limited to 100 characters. The workflow name composes of 3 main parts:
- username
- RequestString
- data/time of submission; This part is added during the submission, max. characters are 19, e.g. `_200425_064221_5181`.

This PR is to check if len(username_RequestString) is longer than 81 characters or not, then print the warning at the end when `--wm init` is used. 

The goal is not to break the `--wm init` process, as one may need to just dump config for testing using any release (including IB). This PR just provides the warning in case someone would like to inject workflows later. Without this test, the injection will break when `--wm force` is used. However, some of the workflows will be submitted if they come before problematic workflows.

#### PR validation:
Test with 11_1_0_pre6:
runTheMatrix.py --what standard -l 136.886,136.7611,136.756 -t 8 -b 'DQM8CoresData' --label 'DQM8CoresData1234567890' --wm init

```
*** FIX NEEDED BEFORE INJECTION: 2 candidate workflows have too long name (>81 characters) ***
srimanob_RVCMSSW_11_1_0_pre6RunJetHT2016E_reminiaod__DQM8CoresData1234567890_RelVal_2016Ermaod (94 characters) 
srimanob_RVCMSSW_11_1_0_pre6RunZeroBias2016D__DQM8CoresData1234567890_RelVal_2016D (82 characters) 
```

workflow 136.866 is OK in this test,
> srimanob_RVCMSSW_11_1_0_pre6RunEGamma2018D__DQM8CoresData1234567890_RelVal_2018D (80 characters long)

During injection, 136.866 will pass the injection, then break when 136.7611 injection is done.

#### if this PR is a backport please specify the original PR and why you need to backport that PR:
No need to backport.

### CODE REVIEW

**Comment by chayanit** [comment]
please test

File: Configuration/PyReleaseValidation/python/MatrixInjector.py [hunk 1]
```diff
@@ -59,7 +59,7 @@ def __init__(self,opt,mode='init',options=''):
         self.batchTime = str(int(time.time()))
         if(opt.batchName):
             self.batchName = '__'+opt.batchName+'-'+self.batchTime
-
+
```
Comment (kpedro88) [inline Configuration/PyReleaseValidation/python/MatrixInjector.py:62]: remove unnecessary whitespace

File: Configuration/PyReleaseValidation/python/MatrixInjector.py [hunk 2]
```diff
@@ -83,7 +83,8 @@ def __init__(self,opt,mode='init',options=''):
         self.speciallabel=''
         if opt.label:
             self.speciallabel= '_'+opt.label
-
+        self.countLongWFName = 0
+        self.longWFName = ''
```
Comment (kpedro88) [inline Configuration/PyReleaseValidation/python/MatrixInjector.py:87]: make this a list, then the separate "count" is unneeded

File: Configuration/PyReleaseValidation/python/MatrixInjector.py [hunk 3]
```diff
@@ -477,6 +478,11 @@ def prepare(self,mReader, directories, mode='init'):
                     chainDict['RequestString']='RV'+chainDict['CMSSWVersion']+s[1].split('+')[0]
                     if processStrPrefix or thisLabel:
                         chainDict['RequestString']+='_'+processStrPrefix+thisLabel
+                    #check candidate WF name
+                    self.candidateWFName = self.user+'_'+chainDict['RequestString']
+                    if (len(self.candidateWFName)>81):
```
Comment (kpedro88) [inline Configuration/PyReleaseValidation/python/MatrixInjector.py:483]: magic number 81 should be named constant

File: Configuration/PyReleaseValidation/python/MatrixInjector.py [hunk 4]
```diff
@@ -477,6 +478,11 @@ def prepare(self,mReader, directories, mode='init'):
...
+                        self.countLongWFName+=1
+                        self.longWFName+=self.candidateWFName+' ('+str(len(self.candidateWFName))+' characters) \n'
```
Comment (kpedro88) [inline Configuration/PyReleaseValidation/python/MatrixInjector.py:485]: just append to list

File: Configuration/PyReleaseValidation/python/MatrixInjector.py [hunk 5]
```diff
@@ -647,6 +653,6 @@ def submit(self):
                 workFlow=makeRequest(self.wmagent,d,encodeDict=True)
                 print("...........",n,"submitted")
                 random_sleep()
-            
-
-        
+        if self.testMode and self.countLongWFName>0:
+            print("\n*** FIX NEEDED BEFORE INJECTION: "+str(self.countLongWFName)+" candidate workflows have too long name (>81 characters) ***")
```
Comment (kpedro88) [inline Configuration/PyReleaseValidation/python/MatrixInjector.py:657]: raise an exception?

File: Configuration/PyReleaseValidation/python/MatrixInjector.py [hunk 6]
```diff
@@ -647,6 +653,6 @@ def submit(self):
...
+            print(self.longWFName)
```
Comment (kpedro88) [inline Configuration/PyReleaseValidation/python/MatrixInjector.py:658]: `print('\n'.join(self.longWFName))` (once converted to a list)

File: Configuration/PyReleaseValidation/python/MatrixInjector.py (hunk 1 above)
Comment (srimanob) [reply Configuration/PyReleaseValidation/python/MatrixInjector.py]: Fixed

File: Configuration/PyReleaseValidation/python/MatrixInjector.py (hunk 2 above)
Comment (srimanob) [reply Configuration/PyReleaseValidation/python/MatrixInjector.py]: Fixed, make the list instead.

File: Configuration/PyReleaseValidation/python/MatrixInjector.py (hunk 3 above)
Comment (srimanob) [reply Configuration/PyReleaseValidation/python/MatrixInjector.py]: Done, define a constant MAXWORKFLOWLENGTH at the beginning of the file.

File: Configuration/PyReleaseValidation/python/MatrixInjector.py (hunk 4 above)
Comment (srimanob) [reply Configuration/PyReleaseValidation/python/MatrixInjector.py]: Fixed.

File: Configuration/PyReleaseValidation/python/MatrixInjector.py (hunk 5 above)
Comment (srimanob) [reply Configuration/PyReleaseValidation/python/MatrixInjector.py]: Fix the text. This is for warning only that the workflows can't be submitted later.

File: Configuration/PyReleaseValidation/python/MatrixInjector.py (hunk 6 above)
Comment (srimanob) [reply Configuration/PyReleaseValidation/python/MatrixInjector.py]: Done. Thanks.

**Comment by chayanit** [comment]
please test

File: Configuration/PyReleaseValidation/python/MatrixInjector.py (hunk 5 above)
Comment (kpedro88) [reply Configuration/PyReleaseValidation/python/MatrixInjector.py]: Is there a situation where we want to allow those long workflow names? If not, I still think an exception would be better. That way, any PR creating an unnecessarily long workflow name must be fixed.

Comment (srimanob) [reply Configuration/PyReleaseValidation/python/MatrixInjector.py]: As mentioned in the PR description, if someone would like to use "--wm init" to dump the configuration to run local-test only, we should not break it. This warning is to target for people who will submit relvals later, using "--wm submit, force" after "--wm init".

**Comment by silviodonato** [comment]
@kpedro88 does the last version of the code address your comments?
 @chayanit @pgunnell @wajidalikhan do you have any comments?

**Comment by chayanit** [comment]
+1

**Comment by kpedro88** [comment]
+upgrade

**Comment by silviodonato** [comment]
+1
